### ISSUE TITLE
Starting a transition while a transition is running creates a ghost snapshot

### ISSUE THREAD

**Issue by Owumaro** [cristianbote/phaser-state-transition#23]
If you start a transition while a transition is already running, there is a ghost snapshot that doesn't disappear properly. Here is an example :

https://jsfiddle.net/Owumaro/a9xvn1jb/5/

I have 3 states that start in cycle (1, 2, 3) by pressing the enter key. The transition lasts 1 second, so you can easily hit enter while a transition is already running.

It looks like the end snapshot of the 1st transition goes on top at the end of the 2nd transition.

**Comment by cristianbote** [comment]
Hey, thanks for submitting this issue.

Indeed, something weird is going on. I'll get back to you, as soon as I have a chance of debugging this.
Meanwhile, if you already know the fix, feel free to submit a pull request.

Cheers!

**Comment by Owumaro** [comment]
Well it's pretty hard for me to debug this as I don't understand all the mechanics of the plugin yet.

From my investigation, here is what I found :
* in `Slide.prototype.go`, a `ContentSnapshot` is added to the `game.stage`
* then, the `Transition.start` method is called which creates a tween for the snapshot, and adds an `onComplete` callback on this tween that removes the tween and the snapshot (via `this.currentTarget.destroy()`)

However, if a second transition starts before the tween completes, the tween of the first transition is interrupted (and its `onComplete` callbacks aren't called). Therefore the tween has been destroyed (by Phaser itself when the state changed ?), however the snapshot is still here, stuck in the middle of the tween.

Sadly there is no event on `Phaser.tween` when it gets removed... The only solution I found is to use the `Phaser.StateManager.stateChange` event :

```
this.game.state.onStateChange.addOnce(function() {
        this.currentTarget.destroy();
}, this)
```
(set right after the [tween `onComplete` callback set](https://github.com/cristianbote/phaser-state-transition/blob/c833798e2401bb6c5530b708cdb502897b0a79aa/src/transition/Transition.js#L79))

It works but it isn't very clean.
